shardingsphere-infra/shardingsphere-infra-binder/src/main/java/org/apache/shardingsphere/infra/binder/statement/dml/InsertStatementContext.java

```diff
@@ -62,38 +65,49 @@ public final class InsertStatementContext extends CommonSQLStatementContext<Inse
     
     private final List<String> columnNames;
     
+    private final Map<Integer, List<String>> columnNamesMap = new LinkedHashMap<>();
+    
     private final Map<String, ShardingSphereMetaData> metaDataMap;
     
     private final String defaultDatabaseName;
     
     private final List<String> insertColumnNames;
     
-    private final List<List<ExpressionSegment>> valueExpressions;
+    private final Map<Integer, List<List<ExpressionSegment>>> valueExpressions = new LinkedHashMap<>();
     
     private List<InsertValueContext> insertValueContexts;
     
+    private Map<Integer, List<InsertValueContext>> insertValueContextsMap = new LinkedHashMap<>();
+    
     private InsertSelectContext insertSelectContext;
     
     private OnDuplicateUpdateContext onDuplicateKeyUpdateValueContext;
     
-    private GeneratedKeyContext generatedKeyContext;
+    private Map<Integer, GeneratedKeyContext> generatedKeyContexts = new LinkedHashMap<>();
     
     public InsertStatementContext(final Map<String, ShardingSphereMetaData> metaDataMap, final List<Object> parameters,
                                   final InsertStatement sqlStatement, final String defaultDatabaseName) {
         super(sqlStatement);
         this.metaDataMap = metaDataMap;
         this.defaultDatabaseName = defaultDatabaseName;
         insertColumnNames = getInsertColumnNames();
-        valueExpressions = getAllValueExpressions(sqlStatement);
-        AtomicInteger parametersOffset = new AtomicInteger(0);
-        insertValueContexts = getInsertValueContexts(parameters, parametersOffset, valueExpressions);
-        insertSelectContext = getInsertSelectContext(metaDataMap, parameters, parametersOffset, defaultDatabaseName).orElse(null);
-        onDuplicateKeyUpdateValueContext = getOnDuplicateKeyUpdateValueContext(parameters, parametersOffset).orElse(null);
         tablesContext = new TablesContext(getAllSimpleTableSegments(), getDatabaseType());
         ShardingSphereSchema schema = getSchema(metaDataMap, defaultDatabaseName);
-        columnNames = useDefaultColumns() ? schema.getAllColumnNames(sqlStatement.getTable().getTableName().getIdentifier().getValue()) : insertColumnNames;
-        generatedKeyContext = new GeneratedKeyContextEngine(sqlStatement, schema)
-                .createGenerateKeyContext(insertColumnNames, getAllValueExpressions(sqlStatement), parameters).orElse(null);
+        AtomicInteger parametersOffset = new AtomicInteger(0);
+        List<InsertStatement> insertStatements = getInsertStatements();
+        for (int index = 0; index < insertStatements.size(); index++) {
+            InsertStatement insertStatement = insertStatements.get(index);
```

Comment:
Please don't use `get(index)` on LinkedList, its performance will be very poor, I think an extra counter can be used instead of index.